fortuna/data/dataset/huggingface_datasets.py

```diff
@@ -0,0 +1,186 @@
+        self.total_train_batch_size = per_device_train_batch_size * jax.local_device_count()
+        self.total_eval_batch_size = per_device_eval_batch_size * jax.local_device_count()
+        self._data_collator = None
+
+    @abc.abstractmethod
+    def get_tokenized_datasets(self, datasets: DatasetDict, text_columns: Sequence[str] = ('sentence',), target_column: str = 'label') -> DatasetDict:
+        """
+        Converts one or more dataset entries into sequences of tokens, using the tokenizer.
+
+        Parameters
+        ----------
+        datasets: DatasetDict
+            A dictionary of `datasets.Dataset` that have to be encoded.
+        text_columns: str
+            A list containing the text column names, whose text sequences has to be encoded.
+        target_column: str
+            The target column name
+
+        Returns
+        -------
+        DatasetDict
+            A dictionary of tokenized datasets.
+        """
+        pass
+
+    @property
+    @abc.abstractmethod
+    def data_collator(self):
+        """
+        Create a data collator object that will process batches of data and apply transformation on them if needed.
+        """
+        pass
+
+    @abc.abstractmethod
+    def _collate(self, batch):
+        pass
+
+    def _get_batches_idxs(self, rng: PRNGKeyArray, dataset_size: int, batch_size: int, shuffle: bool =False, drop_last: bool = False) -> Iterable[Array]:
+        if shuffle:
+            dataset_idxs = jax.random.permutation(rng, dataset_size)  # batch idxs
+        else:
+            dataset_idxs = jnp.arange(dataset_size)
+        if drop_last:
+            steps_per_epoch = dataset_size // batch_size
+            dataset_idxs = dataset_idxs[: steps_per_epoch * batch_size]  # Skip incomplete batch.
+        yield from [dataset_idxs[start:start+batch_size] for start in range(0, len(dataset_idxs), batch_size)]
+
+    def _get_data_loader(
+            self, dataset: Dataset,  shuffle: bool = False, drop_last: bool = False, rng: Optional[PRNGKeyArray] = None
+    ) -> Union[Iterable[Dict[str, Array]], Iterable[Tuple[Dict[str, Array],Array]]]:
+        self.rngs['train'] = jax.random.split(self.rngs['train'], 1)[0]
+
+        batch_idxs_gen = self._get_batches_idxs(
+            rng=rng,
+            dataset_size=len(dataset),
+            batch_size=self.total_train_batch_size,
+            shuffle=shuffle,
+            drop_last=drop_last,
+        )
+
+        for batch_idxs in batch_idxs_gen:
+            batch = dataset[batch_idxs]
+            batch = self._collate(batch)
+            batch_inputs = {k: jnp.array(v) for k, v in batch.items()}
+            if 'labels' in batch_inputs:
+                batch_labels = batch_inputs.pop('labels')
+                yield (batch_inputs, batch_labels)
+            else:
+                yield batch_inputs
+
+    def get_train_dataloader(self, dataset: Dataset) -> HuggingFaceDataLoader:
```

Comment:
Related to the comment above, should we consider also offering a `get_val_data_loader` and `get_calib_data_loader`?

Or, more generally, should we consider offering a generic `get_data_loader`, where the user can decide whether to shuffle or not (for example)? I can imagine for output calibration perhaps one might want a data loader that does not shuffle.